Std/Data/BitVec/Basic.lean

```diff
@@ -0,0 +1,259 @@
+/-
+Copyright (c) 2022 by the authors listed in the file AUTHORS and their
+institutional affiliations. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joe Hendrix, Wojciech Nawrocki, Leonardo de Moura, Mario Carneiro
+-/
+import Std.Data.Nat.Init.Lemmas
+import Std.Tactic.Alias
+
+namespace Std
+
+/-!
+We define bitvectors. We choose the `Fin` representation over others for its relative efficiency
+(Lean has special support for `Nat`), alignment with `UIntXY` types which are also represented
+with `Fin`, and the fact that bitwise operations on `Fin` are already defined. Some other possible
+representations are `List Bool`, `{ l : List Bool // l.length = w }`, `Fin w → Bool`.
+
+We also define many bitvector operations from the
+[`QF_BV` logic](https://smtlib.cs.uiowa.edu/logics-all.shtml#QF_BV).
+of SMT-LIBv2.
+-/
+
+/--
+A bitvector of the specified width. This is represented as the underlying `Nat` number
+in both the runtime and the kernel, inheriting all the special support for `Nat`.
+-/
+structure BitVec (w : Nat) where
+  /-- Construct a `BitVec w` from a number less than `2^w`.
+  O(1), because we use `Fin` as the internal representation of a bitvector. -/
+  mk ::
+  /-- Interpret a bitvector as a number less than `2^w`.
+  O(1), because we use `Fin` as the internal representation of a bitvector. -/
+  val : Fin (2^w)
+  deriving DecidableEq
+
+namespace BitVec
+
+/-- The `BitVector` with value `i mod 2^n`. Treated as an operation on bitvectors,
```

Comment:
This docstring says `BitVector` (in backticks) while the type is `BitVec`. I don't intend to do any name bikeshedding here, but presumably one way or the other this is a remnant of an incomplete rename that should be concluded?

Reply:
That's my fault, it was written without backticks but with camel case and I added the backticks.
```suggestion
/-- The `BitVec` with value `i mod 2^n`. Treated as an operation on bitvectors,
```